### PR TITLE
PERF: Improve performance for astype is view

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

This came up in asvs, it wasn't much but we can do better

### CODE REVIEW

**Review by jorisvandenbossche — APPROVED** [review]
Looks good

**Comment by jorisvandenbossche** [comment]
(there is a mypy error ...)

File: pandas/core/dtypes/astype.py [hunk 1]
```diff
@@ -258,6 +258,11 @@ def astype_is_view(dtype: DtypeObj, new_dtype: DtypeObj) -> bool:
     -------
     True if new data is a view or not guaranteed to be a copy, False otherwise
     """
+    if dtype.kind in "iufb" and dtype.kind == new_dtype.kind:
```
Comment (mroeschke) [inline pandas/core/dtypes/astype.py:261]: `int64` to `Int64` (numpy to masked) is okay to short cut here?

Comment (phofl) [reply pandas/core/dtypes/astype.py:261]: yeah this is definitely a view

File: pandas/core/dtypes/astype.py [hunk 2]
```diff
@@ -258,6 +258,11 @@ def astype_is_view(dtype: DtypeObj, new_dtype: DtypeObj) -> bool:
...
+        # fastpath for numeric dtypes
+        if hasattr(dtype, "itemsize") and hasattr(new_dtype, "itemsize"):
+            return dtype.itemsize == new_dtype.itemsize  # type: ignore[reportGeneralTypeIssues]
```
Comment (twoertwein) [inline pandas/core/dtypes/astype.py:264]: When pyright finds an issue but not mypy, you have to use the pyright-specific ignore comment: `# pyright: ignore[report...]`

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**